lib/swagger_yard/operation.rb

```diff
@@ -170,6 +173,15 @@ def sort_parameters
       @parameters.sort_by! {|p| p.name}
     end
 
+    ##
+    # Example:
+    # @extensions [x-internal: true, x-beta: true]
+    def add_extensions(tag)
+      extension_regex = /\[([\w-]*\:\s[\w-]*)(?:\,\s([\w-]*\:\s[\w-]*))*\]/
+      extension_matches = tag.text.match(extension_regex).captures.compact
```

Comment:
Since the regex can return `nil` if it doesn't match, should we check before calling `captures.compact`? Ideally warning the user, I think we can with `SwaggerYard.log.warn`

eg:
```
> extension_regex = /\[([\w-]*\:\s[\w-]*)(?:\,\s([\w-]*\:\s[\w-]*))*\]/
=> /\[([\w-]*\:\s[\w-]*)(?:\,\s([\w-]*\:\s[\w-]*))*\]/
> "[x-internal: true, x-example-url: http://example.com/foo]".match(extension_regex)
=> nil
```

Any particular reason to go with the `[key: value, key: value]` format that I'm missing?

If not, an option could be to have `@extensions` behave like parameters, in where you can specify it multiple times, like
```
@extension x-internal: true
@extension x-beta: true
```
I think that would make the parsing simpler, we can do something like`key, value = tag.text.split(":", 2)` and add the results to the `@extentions` hash if valid. 
Less regex to worry about 😄, and we could also validate/ignore extensions that don't start with `x-`


What do you think?

Reply:
Great suggestions! I do like that better.